### PR TITLE
Avoid throttling of cf describeStackEvents api calls

### PR DESCRIPTION
**Pull request by MarcelBoettcher4007**
Problem: Currently all events are loaded, even old once. As the event
history is long the api calls could page a few times and run into
throttling. Updating multiple stacks in parallel (e.g. on CI) can
easily cause throttling in an endless loop.

Events are loaded sorted by timestamp desc (newest events first).
So we do not need to load all of them, but can stop if we loaded an
event older than the starting time of our stack creation/update.

### CODE REVIEW

**Comment by MarcelBoettcher4007** [comment]
Hi, just want to report that we are using this fix since 5 days and see an 3-10x increase in speed of our stack updates as we avoid throttling of concurrently running jobs. This of course really depends on your case/usage, but the bottom line is: it works and is tested in production use.

Would love to see this merged.

Thanks for you awesome work!
Cheers
Marcel

File: index.js [hunk 1]
```diff
@@ -187,15 +187,21 @@ function Cfn (name, template) {
             .then(function (data) {
               next = (data || {}).NextToken
               allEvents = allEvents.concat(data.StackEvents)
-              return !next ? Promise.resolve() : getStackEvents()
+              return next && !allRelevantEventsLoaded(data.StackEvents) ? getStackEvents() : Promise.resolve()
```
Comment (andyday) [inline index.js]: This smells of a hack.  What are you trying to accomplish?  If the returned event list has events earlier than when we started this deploy?  So don't show any events in this case?

Comment (MarcelBoettcher4007) [reply index.js]: Hi @andyday,
thanks for your review.

The api docs for [describeStackEvents](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/CloudFormation.html#describeStackEvents-property) states: "Returns all stack related events for a specified stack in reverse chronological order. "

So if we see events older than our startedAt date, we know that we have loaded all relevant events and can stop paging.

Does this makes sense to you? Tried to explain it in the comment of allRelevantEventsLoaded.

Cheers
Marcel

**Comment by andyday** [comment]
published in 1.7.0

**Pull request merged**